docs/pages/router/reference/hooks.mdx

```diff
@@ -0,0 +1,168 @@
+---
+title: Hooks API
+description: Learn how to interact with the in-app URL in Expo Router.
+---
+
+import { FileTree } from '~/ui/components/FileTree';
+import { RouteUrlGroup, RouteUrl } from '~/ui/components/RouteUrl';
+
+<RouteUrlGroup>
+
+In Expo Router, there's always a valid URL which represents the currently focused route. Use hooks to observe changes and interact with the URL.
+
+## `usePathname`
+
+Returns the currently selected route location without search parameters. e.g. `/acme?foo=bar` -> `/acme`. Segments will be normalized: `/[id]?id=normal` -> `/normal`
+
+<RouteUrl>/profile/baconbrix?extra=info</RouteUrl>
+
+```js app/profile/[user].tsx
+import { Text } from 'react-native';
+/* @info */
+import { usePathname } from 'expo-router';
+/* @end */
+
+export default function Route() {
+  /* @info <b>pathname = "/profile/baconbrix"</b> */
+  const pathname = usePathname();
+  /* @end */
+
+  return <Text>User: {user}</Text>;
+}
+```
+
+## `useLocalSearchParams`
+
+Returns the URL search parameters for the contextually selected route. Refer to the [local vs global search params](/router/reference/search-parameters/#local-vs-global-search-parameters) guide for more info.
+
+<FileTree files={['app/_layout.tsx', 'app/[first]/home.tsx', 'app/[second]/shop.tsx']} />
+
+When `/abc/home` pushes `/123/shop`, `useGlobalSearchParams` would return `{ first: undefined, second: '123' }` on `/app/[first]/home.tsx` because the global URL has changed. However, you may want the params to remain `{ first: 'abc' }` to reflect the context of the screen. In this case, you can use `useLocalSearchParams` to ensure the params `{ first: 'abc' }` are still returned in `/app/[first]/home.tsx`.
```

Comment:
```suggestion
When `/abc/home` pushes `/123/shop`, `useGlobalSearchParams` returns `{ first: undefined, second: '123' }` on **/app/[first]/home.tsx** because the global URL has changed. However, you may want the params to remain `{ first: 'abc' }` to reflect the context of the screen. In this case, you can use `useLocalSearchParams` to ensure the params `{ first: 'abc' }` are still returned in **/app/[first]/home.tsx**.
```